plugins/chunter-resources/src/components/EditChannel.svelte

```diff
@@ -0,0 +1,144 @@
+    channel = result[0]
+  })
+
+  const tabLabels: IntlString[] = [chunter.string.Channel, chunter.string.Members]
+  let selectedTabIndex = 0
+
+  function onNameChange (ev: Event) {
+    const value = (ev.target as HTMLInputElement).value
+    if (value.trim().length > 0) {
+      client.updateDoc(_class, channel.space, channel._id, { name: value })
+    } else {
+      // Just refresh value
+      query.query(chunter.class.Channel, { _id }, (result) => {
+        channel = result[0]
+      })
+    }
+  }
+
+  function onTopicChange (ev: Event) {
+    const newTopic = (ev.target as HTMLInputElement).value
+    client.updateDoc(_class, chunter.space.Topic, channel._id, { topic: newTopic })
+  }
+
+  function onDescriptionChange (ev: Event) {
+    const newDescription = (ev.target as HTMLInputElement).value
+    client.updateDoc(_class, chunter.space.Description, channel._id, { description: newDescription })
+  }
+</script>
+
+<div
+  class="antiOverlay"
+  on:click={() => {
+    dispatch('close')
+  }}
+/>
+<div class="antiDialogs antiComponent">
+  <div class="ac-header short mirror divide">
+    <div class="ac-header__wrap-title">
+      <div class="ac-header__icon">
+        {#if clazz.icon}<Icon icon={clazz.icon} size={'medium'} />{/if}
+      </div>
+      <div class="ac-header__title"><Label label={clazz.label} /></div>
+    </div>
+    <div class="tool">
+      <ActionIcon
+        icon={IconClose}
+        size={'small'}
+        action={() => {
+          dispatch('close')
+        }}
+      />
+    </div>
+  </div>
+  <div class="ac-tabs">
+    {#each tabLabels as tabLabel, i}
+      <div
+        class="ac-tabs__tab"
+        class:selected={i === selectedTabIndex}
+        on:click={() => {
+          selectedTabIndex = i
+        }}
+      >
+        <Label label={tabLabel} />
+      </div>
+    {/each}
+    <div class="ac-tabs__empty" />
+  </div>
+  <Scroller padding>
+    {#if selectedTabIndex === 0}
+      <!-- Channel description -->
+      {#if channel}
+        <div class="flex-col flex-gap-3">
+          <EditBox
+            label={clazz.label}
+            icon={clazz.icon}
+            bind:value={channel.name}
+            placeholder={clazz.label}
+            maxWidth="39rem"
+            focus
+            on:change={onNameChange}
+          />
+          <EditBox
+            label={chunter.string.Topic}
+            bind:value={channel.topic}
+            placeholder={chunter.string.Topic}
+            maxWidth="39rem"
+            focus
+            on:change={onTopicChange}
+          />
+          <EditBox
+            label={chunter.string.ChannelDescription}
+            bind:value={channel.description}
+            placeholder={chunter.string.ChannelDescription}
+            maxWidth="39rem"
+            focus
+            on:change={onDescriptionChange}
+          />
+          <Attachments objectId={channel._id} _class={channel._class} space={channel.space} />
```

Comment:
Here we need another component to show all space attachments, table for example, please remove it currenlty

Reply:
Done